src/sentry/api/endpoints/organization_release_details.py

```diff
@@ -14,18 +18,168 @@
 )
 from sentry.models import Activity, Project, Release, ReleaseCommitError
 from sentry.models.release import UnsafeReleaseDeletion
-from sentry.snuba.sessions import STATS_PERIODS, get_release_sessions_time_bounds
+from sentry.snuba.sessions import (
+    STATS_PERIODS,
+    get_adjacent_releases_based_on_adoption,
+    get_release_sessions_time_bounds,
+)
 from sentry.utils.sdk import bind_organization_context, configure_scope
 
 
+class InvalidSortException(Exception):
+    pass
+
+
 class OrganizationReleaseSerializer(ReleaseSerializer):
     headCommits = ListField(
         child=ReleaseHeadCommitSerializerDeprecated(), required=False, allow_null=False
     )
     refs = ListField(child=ReleaseHeadCommitSerializer(), required=False, allow_null=False)
 
 
-class OrganizationReleaseDetailsEndpoint(OrganizationReleasesBaseEndpoint, ReleaseAnalyticsMixin):
+class OrganizationReleaseDetailsPaginationMixin:
+    @staticmethod
+    def __get_prev_release_date_query_q_and_order_by(release):
+        """
+        Method that takes a release and returns a dictionary containing a date query Q expression
+        and order by columns required to fetch previous release to that passed in release on date
+        sorting
+        """
+        return {
+            "date_query_q": [
+                Q(date_added__gte=release.date_added),
+                Q(date_added__gt=release.date_added) | Q(id__gt=release.id),
```

Comment:
This took me a few reads to understand. Would this be clearer as:
`Q(date_added=release.date_added, id__gt=release.id) | Q(date_added__gt=release.date_added)`

Reply:
Yeah you are right! I was falsely under the impression that clause order plays a role as in "first where clause will filter all gte entries and then second Q expression will just filter down further for the ones that have a date that is either greater or has a greater id" but I looked into it and its not true so changed it!